src/component/TodoInput.jsx

```diff
@@ -1,27 +1,35 @@
-import { useState } from "react";
+import { array, func, string } from "prop-types"
 
-export default function TodoInput({ todos, setTodos, setflag ,task,setTask }) {
-  
+export default function TodoInput({ todos, setTodos, todo, setTodo }) {
+    const handleSubmit = (e) => {
+        e.preventDefault()
+        const value = e.currentTarget.elements.newTodo.value.trim()
+        if (!value) {
+            return
+        }
 
-  function handleTodos() {
-    if (!task) return;
-    setTodos([...todos, task]);
-    setTask("");
-    setflag(true);
-  }
-  function handleTodoEnter(e) {
-    e.key === "Enter" ? handleTodos() : "";
-  }
-  console.log(task);
-  return (
-    <header>
-      <input
-        placeholder="Enter todo..."
-        value={task}
-        onChange={(e) => setTask(e.target.value)}
-        onKeyDown={(e) => handleTodoEnter(e)}
-      />
-      <button onClick={handleTodos}>Add</button>
-    </header>
-  );
+        setTodos([...todos, value])
+        setTodo("")
+    }
+
+    return (
+        <form onSubmit={handleSubmit}>
+            <input
+                id="newTodo"
+                type="text"
+                placeholder="Enter todo..."
+                value={todo}
+                onChange={(e) => setTodo(e.target.value)}
+            />
+
+            <button type="submit">Add</button>
+        </form>
+    )
+}
+TodoInput.displayName = "TodoInput"
+TodoInput.propTypes = {
```

Comment:
PropTypes validation to:
- Give an indications for other engineers on the values that should be passed as props
- Make sure that we are passing the correct values (React will throw an error on development mode on prop validation failure)